### PR TITLE
backend: Encode state in base64, use this rather than plain cluster name

### PR DESCRIPTION
**Pull request by ashu8912**
## Testing done
Testing not done yet, this would be an opportunity to test with the recent oidc fixes on keycloak and dex as well

### CODE REVIEW

**Comment by ashu8912** [comment]
fixes #440

**Review by joaquimrocha — COMMENTED** [review]
> State in oidc provides us security

I find this comment a bit confusing since we're talking about encoding/decoding and security here and not exposing the state, which sounds like encrypting it but it's of course not what this is about. So I'd change the message and say something like: "The OIDC state is used for verifying the incoming OIDC callback requests and we should make sure it's transmitted without issues, regardless of its value, so this patch starts using that state in encoded in base64. This also fixes an issues related to the state's length, as it needs to be longer than 8 chars, and this new base64 encoding grants that."

File: backend/cmd/headlamp.go [hunk 1]
```diff
@@ -307,14 +308,20 @@ func StartHeadlampServer(config *HeadlampConfig) {
...
+		*/
+		state := base64.StdEncoding.EncodeToString([]byte(cluster))
+		oauthRequestMap[state] = &OauthConfig{Config: oauthConfig, Verifier: verifier, Ctx: ctx}
+		http.Redirect(w, r, oauthConfig.AuthCodeURL(state), http.StatusFound)
 	}).Queries("cluster", "{cluster}")
 
 	r.HandleFunc("/oidc-callback", func(w http.ResponseWriter, r *http.Request) {
 		state := r.URL.Query().Get("state")
+		decodedState, err := base64.StdEncoding.DecodeString(state)
+		if err != nil {
```
Comment (joaquimrocha) [inline backend/cmd/headlamp.go:322]: We should probably refer this error in the HTTP error and/or log it, so eventual bug reports are more complete.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
Left a comment about logging/returning an error relating to decoding.
After that we can merge this.

File: backend/cmd/headlamp.go (hunk 1 above)
Comment (ashu8912) [reply backend/cmd/headlamp.go:322]: fixed

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**